core/src/main/java/hudson/tools/DownloadFromUrlInstaller.java

```diff
@@ -148,6 +146,23 @@ public String getId() {
             if(d==null)     return Collections.emptyList();
             return Arrays.asList(((InstallableList)JSONObject.toBean(d,InstallableList.class)).list);
         }
+        
+        /**
+         * The UI uses this information to populate the drop-down.
+         * 
+         * @return JSON array of installable tools, never null
+         * @since 1.617
+         */
+        public JSONArray getInstallablesJson() throws IOException {
```

Comment:
I am not sure what the original intention was but the javadoc does indeed sound misleading for the use-cases we use it for.

Generally, we try to expose as little API as possible. And as stapler/jelly require public access modifier in several places, `@Restricted` is the second best thing. `DoNotUse` for what is accessed via reflection only, `NoExternalUse` for what needs to be referenced from the project only (tests usually).